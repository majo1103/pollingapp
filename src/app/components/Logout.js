import React from "react";

import { Redirect } from "react-router-dom";

import { client } from "../Client";

export class Logout extends React.Component {
  constructor(props) {
    super(props);

    client.logout();
  }

  render() {
    return <Redirect to="/login" />;
  }
}
