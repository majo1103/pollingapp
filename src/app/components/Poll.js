import React from "react";

export class Poll extends React.Component {
  render() {
    return (
      <h1 className="text-center">
        What is your favorite programming language?
      </h1>
    );
  }
}
