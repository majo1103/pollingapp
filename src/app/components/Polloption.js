import React from "react";
import { runInDebugContext } from "vm";

export class Polloption extends React.Component {
  constructor(props) {
    super(props);
    this.handleUpVote = this.handleUpVote.bind(this);
  }

  handleUpVote() {
    this.props.onChange(this.props.id);
  }

  render() {
    return (
      <div>
        <input
          type="radio"
          className="form-check-input"
          checked={this.props.active}
          onChange={this.handleUpVote}
        />
        <label className="form-check-label" style={{ color: "white" }}>
          {this.props.title}
        </label>
      </div>
    );
  }
}
