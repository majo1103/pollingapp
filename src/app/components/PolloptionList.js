import React from "react";
import { Polloption } from "./Polloption";
import { options } from "../../seed";

export class PolloptionList extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      options: []
    };
    this.handleOptionUpVote = this.handleOptionUpVote.bind(this);
  }

  componentDidMount() {
    this.setState({ options: options });
  }

  handleOptionUpVote(optionId) {
    const nextOptions = this.state.options.map(option => {
      if (option.id === optionId) {
        return Object.assign({}, option, {
          votes: option.votes + 1
        });
      } else {
        return option;
      }
    });
    this.setState({
      options: nextOptions
    });
  }
  render() {
    const polloptions = this.state.options.map(option => (
      <Polloption
        key={"option-" + option.id}
        id={option.id}
        title={option.title}
        votes={option.votes}
        onVote={this.handleOptionUpVote}
      />
    ));
    return <ul>{polloptions}</ul>;
  }
}
