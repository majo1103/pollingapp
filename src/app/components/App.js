import React from "react";
import { PrivateRoute } from "./PrivateRoute";
import { PolloptionList } from "./polloptionList/PolloptionList";
import { Logout } from "./Logout";
import { Login } from "./login/Login";
import { Register } from "./register/Register";
import { Route, Redirect, Switch } from "react-router-dom";

const NoMatch = ({ location }) => (
  <div>
    <strong>Error!</strong> No route found matching:
    <div>
      <code>{location.pathname}</code>
    </div>
  </div>
);

export class App extends React.Component {
  render() {
    return (
      <Switch>
        <PrivateRoute path="/poll" component={PolloptionList} />
        <Route path="/login" component={Login} />
        <Route path="/logout" component={Logout} />
        <Route path="/register" component={Register} />
        <Route exact path="/" render={() => <Redirect to="/poll" />} />
        <Route component={NoMatch} />
      </Switch>
    );
  }
}
