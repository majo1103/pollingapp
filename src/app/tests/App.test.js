import { App } from "../components/App";
import React from "react";
import { shallow } from "enzyme";

describe("App", () => {
  it("should have the `switch` element", () => {
    const wrapper = shallow(<App />);
    expect(wrapper).toMatchSnapshot();
  });
});
