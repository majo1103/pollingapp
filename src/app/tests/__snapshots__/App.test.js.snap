// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should have the \`switch\` element 1`] = `
<Switch>
  <PrivateRoute
    component={[Function]}
    path="/poll"
  />
  <Route
    component={[Function]}
    path="/login"
  />
  <Route
    component={[Function]}
    path="/logout"
  />
  <Route
    component={[Function]}
    path="/register"
  />
  <Route
    exact={true}
    path="/"
    render={[Function]}
  />
  <Route
    component={[Function]}
  />
</Switch>
`;
